### ISSUE TITLE
Great add-on but slight problem with ts

### ISSUE THREAD

**Issue by AskMP** [eliasm307/storybook-addon-deep-controls#1]
argTypes is a "Partial<ArgTypes<TArgs>>" type object which doesn't support quoted deep linked attributes. I'm wondering what would be the best manner in which to append that type other than simply adding "|any" to the end of the storybook types description file (line 1154 of storybook->types->dist->index.d.ts).

**Comment by eliasm307** [comment]
Hi @AskMP thanks for the feedback however I'm having trouble trying to understand the issue.

None of the types used in this addon are exposed to apps that use the addon, there is just one parameter which is a boolean, so it doesnt matter what `argTypes` type is used in the addon? Please let me know if I'm missing something

> quoted deep linked attributes
Could you please clarify what this means? Do you have some example code?

> best manner in which to append that type other than simply adding "|any" to the end of the storybook types description file (line 1154 of storybook->types->dist->index.d.ts)

Are you able to provide a link or a picture or something to the file you are referring to?

**Comment by shaheemMPM** [comment]
I'm encountering a similar issue when using this addon with TypeScript.

Here's an example of how my Storybook configuration looks:

```javascript
export default {
  ...
  parameters: {
    deepControls: { enabled: true },
  },
  args: {
    editingMode: false,
    showSlNo: true,
    theme: {
      color: "#029CFD",
    },
  },
  argTypes: {
    editingMode: {
      options: [true, false],
      control: { type: "radio" },
      description: "Switch between editing and preview mode",
      table: {
        type: { summary: "boolean" },
        defaultValue: { summary: false },
      },
    },
    showSlNo: {
      options: [true, false],
      control: { type: "radio" },
      description: "Show serial number column",
      table: {
        type: { summary: "boolean" },
        defaultValue: { summary: false },
      },
    },
    "theme.color": {
      control: {
        type: "color",
        presetColors: ["red", "green"],
      },
    },
  },
  ...
};
```

In this configuration, you'll notice that I've used // @ts-ignore to achieve the desired output.

**Comment by eliasm307** [comment]
Hi @shaheemMPM thanks for the comment however in your code example you dont use `// @ts-ignore`, so I'm confused? Which part gives you an issue?

**Comment by shaheemMPM** [comment]
@eliasm307 My apologies for the confusion. It was indeed a mistake on my part. I've updated the initial comment to include the // @ts-ignore line. Thank you for your understanding.

**Comment by eliasm307** [comment]
@shaheemMPM Thanks for the details. I've published v0.3.0 of the package with a type utility to help with this, I've added information on how to use it in the readme ie see https://github.com/eliasm307/storybook-addon-deep-controls#typescript

Let me know if you still have issues
